crates/prune/prune/src/segments/account_history.rs

```diff
@@ -64,34 +66,48 @@ impl<DB: Database> Segment<DB> for AccountHistory {
         }
 
         let mut last_changeset_pruned_block = None;
+        // Deleted account changeset keys (account addresses) with the highest block number deleted
+        // for that key
+        let mut highest_deleted_accounts = FxHashMap::default();
         let (pruned_changesets, done) = provider
             .prune_table_with_range::<tables::AccountChangeSets>(
                 range,
                 &mut limiter,
                 |_| false,
-                |row| last_changeset_pruned_block = Some(row.0),
+                |(block_number, account)| {
+                    highest_deleted_accounts.insert(account.address, block_number);
+                    last_changeset_pruned_block = Some(block_number);
+                },
```

Comment:
is there a bound on how big this hashmap can grow?

so that we can guarantee it won't cause oom.

otherwise, we should be doing this in chunks: 
- delete chunk A, max 1k entries in `highest_deleted_accounts`, change sets
- delete chunk A indices
- delete chunk B changesets
- delete chunk B indices
...

Reply:
yep, it's limited by `prune_delete_limit * blocks_since_last_run / 2`, and with current default it's usually `3500 * 5 / 2`, so 8750 entries. Each entry is `160 bit + 256 bit + 64 bit`, so the total size should be up to 0.5MB + some hashmap overhead.

`blocks_since_last_run` is additionally limited by the `max_reorg_depth`, so we should be fine here https://github.com/paradigmxyz/reth/blob/ed53e2ebcaab641ab32002f86bae6bc4d6a78c6b/crates/prune/prune/src/pruner.rs#L111-L126

Reply:
added a comment https://github.com/paradigmxyz/reth/pull/9312/commits/cf205f7f8e89b3df615c61db66f171c9676bfd19

Reply:
background on where the /2 comes from is needed to be truly helpful

Reply:
yep, added